### PR TITLE
Added unit tests and story for SignatureRequestData component

### LINKED ISSUE #16609: Add unit test and story to new components
## Background
https://github.com/MetaMask/metamask-extension/pull/16018#pullrequestreview-1174267059

### PR DESCRIPTION
**Pull request by VSaric**
## Explanation

Added unit tests and story for `SignatureRequestData` component

* Fixes #16609 

## Screenshots/Screencaps

https://user-images.githubusercontent.com/92527393/206702516-bf2f4f8d-fcb0-4dac-bb0c-8fabafdbfc77.mov

## Manual Testing Steps

This component can be tested by running storybook (`yarn storybook`) and then go to the `COMPONENTS` tab and open in `App/SignatureRequest/SignatureRequestData`.

### CODE REVIEW

**Comment by mirjanaKukic** [comment]
Verified by QA

**Review by brad-decker — DISMISSED** [review]

**Review by darkwing — CHANGES_REQUESTED** [review]
This is what I'm seeing:

Default

Is there possible a CSS file to be added somewhere?

**Comment by VSaric** [comment]
@darkwing You can check story from `signature-request-message.js` file, this component is used there. If I add CSS file for this component I think it will broke the design for SignTypedData v4 signatures.  [Here](https://github.com/MetaMask/metamask-extension/pull/16018#pullrequestreview-1174267059) is the @georgewrmarshall  comment for this. 

> This is what I'm seeing:
> 
> Default
> 
> Is there possible a CSS file to be added somewhere?

**Comment by georgewrmarshall** [comment]
I understand @darkwing's observation haha. It does seem a little out of context and maybe quite fragmented? My reasonings for suggesting creating a story for this component and my opinion with UI is that every component no matter how small should have a story. I believe it will help with discovery and enable it to be tested and worked on in isolation.

File: ui/components/app/signature-request/signature-request-data/signature-request-data.stories.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import SignatureRequestData from './signature-request-data';
+
+export default {
+  title: 'Components/App/SignatureRequestData',
```
Comment (georgewrmarshall) [inline ui/components/app/signature-request/signature-request-data/signature-request-data.stories.js:5]: Title should follow folder structure after `ui/` so if the story lives in  `ui/components/app/signature-request/signature-request-data/signature-request-data.stories.js` then the title should be
```suggestion
  title: 'Components/App/SignatureRequest/SignatureRequestData',
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Story looks good! One suggestion regarding story title and navigation

File: ui/components/app/signature-request/signature-request-data/signature-request-data.stories.js (hunk 1 above)
Comment (georgewrmarshall) [reply ui/components/app/signature-request/signature-request-data/signature-request-data.stories.js:5]: If you ever need storybook boiler plate there is a quick example in the storybook docs. I use it quite a lot 😁  https://metamask.github.io/metamask-storybook/?path=/story/getting-started-documentation-guidelines--page#creating-a-story

Screenshot 2022-12-08 at 11 22 09 AM

Comment (VSaric) [reply ui/components/app/signature-request/signature-request-data/signature-request-data.stories.js]: @georgewrmarshall Done [here](https://github.com/MetaMask/metamask-extension/pull/16759/commits/50f1359cb4bb5220486518775005f7d4e9993c8d).

**Review by georgewrmarshall — APPROVED** [review]
Lgtm!

**Comment by VSaric** [comment]
@darkwing @brad-decker Can I get re-review on this?

**Review by jpuri — APPROVED** [review]
Looks good

**Review by darkwing — APPROVED** [review]

**Pull request merged**